tests/fixtures/configSchemaExtensions/test-plugin.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+class TestPlugin {
+  constructor(serverless) {
+    serverless.configSchemaHandler.defineProvider('someProvider', {
+      provider: {
+        properties: {
+          stage: { type: 'string' },
+          variableSyntax: { type: 'string' },
+          versionFunctions: { type: 'boolean' },
+          remoteFunctionData: { type: 'null' },
```

Comment:
>    * Add a small fix to AWS functions compiler, and ensure this line happens only if we're in context of AWS provider: https://github.com/serverless/serverless/blob/57e4212671ea3027fab9482e6006933e4c5b6c55/lib/plugins/aws/package/compile/functions/index.js#L25
>       (there's a small chance it may break for some custom providers, assuming they also rely on `versionFunctions` but I think it's very unlikely, our excuse is that they're relying on unintended side effect)
>    * Remove this line https://github.com/serverless/serverless/blob/57e4212671ea3027fab9482e6006933e4c5b6c55/lib/plugins/aws/package/compile/functions/index.js#L25

Is there a typo here? You're suggesting to wrap this line in an `if` statement, and then suggesting to remove this very line. Did you mean `remoteFunctionData` somewhere?

> 2. I created a provider with non of above properties in its schema - in result validation fails on above four properties. Still only error mentioning first property is reported to me, while I believe all four should be reported

I believe this is proper behavior. Once AJV ensures there is an error, it immediately reports it. That's why there's almost always only one error message. The only exeption to this pattern is for `anyOf` keyword, when AJV needs to make sure all subschemas fail. It then repotrts multiple errors for every failed subschema that was inside `anyOf` array.

Reply:
> Is there a typo here? You're suggesting to wrap this line in an if statement, and then suggesting to remove this very line. Did you mean remoteFunctionData somewhere?

Sorry by mistake I've pasted same link twice. Second link should be: https://github.com/serverless/serverless/blob/57e4212671ea3027fab9482e6006933e4c5b6c55/lib/plugins/aws/deployFunction/index.js#L22

> Once AJV ensures there is an error, it immediately reports it.

Ok, so it's AJV that reports error only on first invalid property?

Reply:
> Ok, so it's AJV that reports error only on first invalid property?

By default, yes. There is also an option in AJV to enable `allErrors` which is false by default. I tried to enable it, and got lots of weird messages which obviously needs additional fine-tuning:
<img width="888" alt="Screenshot 2020-06-19 at 14 39 50" src="https://user-images.githubusercontent.com/4620130/85129213-65dcb880-b23b-11ea-9c8a-8ffa4611328f.png">

I'd suggest to ship it as is, we may improve it in future PRs

Reply:
I mean we ship with `allErrors` set to default false value, and then improve it in future PRs

Reply:
As for things like `variableSyntax`, `versionFunctions `, etc, I also don't think it is a critical issue. We can update docs and instruct provider implementers to include those options. 

I'm feeling that we are starting to add non-critical features that delay final shipping. And I believe in shipping early so we can gather valuable feedback from users.

What do you think?

Reply:
>  I also don't think it is a critical issue. We can update docs and instruct provider implementers to include those options.

Wouldn't it be easier to patch schema on the go, than documenting that it needs to be done in context of a plugin?

I see it more as design flaw of a framework, which would be nice to handle internally (ideally at some point a feature those options could be implemented as AWS specific - but I wouldn't try to do to that now)

Reply:
I agree, eventually. I was overwhelmed by impatience, and I'm not here to hurry things.

I'll implement this